topology/probes/istio/cluster.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2018 IBM, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package istio
+
+import (
+	"fmt"
+
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/probe"
+	"github.com/skydive-project/skydive/topology/graph"
+	"github.com/skydive-project/skydive/topology/probes/k8s"
+)
+
+const ClusterName = "cluster"
+
+type clusterProbe struct {
+	graph.DefaultGraphListener
+	graph          *graph.Graph
+	clusterIndexer *graph.MetadataIndexer
+}
+
+func dumpCluster(name string) string {
+	return fmt.Sprintf("cluster{'Name': %s}", name)
+}
+
+func (p *clusterProbe) newMetadata(name string) graph.Metadata {
+	return newMetadata("cluster", "", name, nil)
+}
+
+func (p *clusterProbe) addNode(name string) {
+	p.graph.Lock()
+	defer p.graph.Unlock()
+
+	k8s.NewNode(p.graph, graph.GenID(), p.newMetadata(name))
```

Comment:
this it why it's make no sense to export NewNode
you just pollute the k8s namespace

You function didn't do anything different than
p.graph.NewNode(graph.GenID(), p.newMetadata(name))

So use the object form, don't pollute namespace, it's confusing during reading and debugging

That's what I named kind of  "C binding" .... meaning do you plan really do call the function from C ?

Reply:
as explained the reason is that I want to hide the correct setting of `hostID`